api/clients/eigenda_client.go

```diff
@@ -44,21 +44,34 @@ func NewEigenDAClient(log log.Logger, config EigenDAClientConfig) (*EigenDAClien
 	llConfig := NewConfig(host, port, config.ResponseTimeout, !config.DisableTLS)
 	llClient := NewDisperserClient(llConfig, signer)
 
-	codec, err := codecs.BlobEncodingVersionToCodec(config.PutBlobEncodingVersion)
+	lowLevelCodec, err := codecs.BlobEncodingVersionToCodec(config.PutBlobEncodingVersion)
 	if err != nil {
 		return nil, fmt.Errorf("error initializing EigenDA client: %w", err)
 	}
 
+	var codec codecs.BlobCodec
+	if config.DisablePointVerificationMode {
+		codec = codecs.NewNoIFFTCodec(lowLevelCodec)
+	} else {
+		codec = codecs.NewIFFTCodec(lowLevelCodec)
+	}
```

Comment:
Knit - the same effect could be achieved via some encoding option type versus having to manage multiple codec types in different files (i.e, ifft, no_ifft)

Reply:
I'm going to leave this as it is for now on the rationale that this option pattern is most useful when you're in a situation where you would otherwise define multiple constructor functions (with different goals). Since we are happy with one constructor function right now it seems like overkill to decompose the initialization of this into separate option functions.